packages/types/src/codec/UInt.ts

```diff
@@ -24,13 +22,13 @@ type BitLength = 8 | 16 | 32 | 64 | 128 | 256;
 //
 // TODO:
 //   - Apart from encoding/decoding we don't actuall keep check on the sizes, is this good enough?
-export default class UInt extends Base<BN> {
+export default class UInt extends BN implements Codec<UInt> {
   private _bitLength: BitLength;
   private _isHexJson: boolean;
 
   constructor (value: AnyNumber = 0, bitLength: BitLength = 64, isHexJson: boolean = true) {
     super(
-      UInt.decode(value)
+      UInt.decode(value).toNumber()
```

Comment:
Cannot use toNumber here - JS numbers are limited to 53 bits (on uints) - rather toString() it into a decimal representation.